### PR TITLE
 Fixed Korean character crash bug on Safari Browser

### PR DESCRIPTION
**Pull request by stegano**
Fixed Korean character crash bug on Safari Browser
Add Korean character test spec
THX :)

### CODE REVIEW

File: backbone.js [hunk 1]
```diff
@@ -1720,7 +1720,10 @@
     // decoded for comparison. `%25` should not be decoded since it may be part
     // of an encoded parameter.
     decodeFragment: function(fragment) {
-      return decodeURI(fragment.replace(/%25/g, '%2525'));
+      fragment = decodeURI(fragment.replace(/%25/g, '%2525'));
+      // If contains Korean then encode hashString - Korean character crash bug on Safari Browser :(
+      fragment = /[ㄱ-힣]/.test(fragment) ? encodeURIComponent(fragment) : fragment;
```
Comment (jdalton) [inline backbone.js]: Has this been [reported to Safari](https://bugs.webkit.org/query.cgi?format=specific&product=WebKit)?

Comment (stegano) [inline backbone.js]: Hi @jdalton
Thank for your reply :)
I saw your replay and reported it
https://bugs.webkit.org/show_bug.cgi?id=165597

Comment (stegano) [inline backbone.js]: This issue appears to have been fixed in a future version of Safari, but it will not work with older (and current) Safari browsers :(

Comment (stegano) [inline backbone.js]: @jdalton
I hope that this feature will approved because almost all users are still uses the old browser version :)

**Comment by ogonkov** [comment]
Harcoding some language-specific characters is looks weird.

**Comment by stegano** [comment]
@ogonkov  Thank you for your good feedback. Let's find another way :D

**Comment by ogonkov** [comment]
I've got similar problem with broken encoding of Russian, and ended with patching `decodeFragment` on my side

**Pull request closed without merging**